cmd/snap/error.go

```diff
@@ -100,6 +100,12 @@ func errorToCmdMessage(snapName string, e error, opts *client.SnapOptions) (stri
 		// arch/channel/revision. Surface that here somehow!
 
 		msg = i18n.G("snap %q not found")
+		if snapName == "" {
```

Comment:
I appreciate both these concerns. I think it needs a revamp, but if it's ok with you i'll leave that revamp for later? I don't think it's terrible to have it in master as is.